mscore/musescore.cpp

```diff
@@ -6314,6 +6314,14 @@ void MuseScore::cmd(QAction* a, const QString& cmd)
                   _styleDlg->setScore(cs);
             _styleDlg->exec();
             }
+	  else if (cmd == "edit-header-footer") {
+			if (!_styleDlg)
+				  _styleDlg = new EditStyle{ cs,this };
+			else
+				  _styleDlg->setScore(cs);
+			_styleDlg->setPage(3);
```

Comment:
Not sure it is possible, but rather than the 'magic number' can't we use a more waterproof solution, deriving the index from the name of that tab somehow?

Reply:
Yes, that's true will try implementing that.

Reply:
Is iterating over the whole _styleDlg->pageList and conparing each item name with 'Header, Footer' a good idea? I'm searching if I can compare it with something concrete.

Reply:
`EditStyle::setPage()` calls `QListWidget::setCurrentRow()`.
Better use `setPage(pageStack->indexOf(EditStyle::PageHeaderFooter))` or something along those lines (read: entirely untested and unsure ;-))?

Reply:
My assumption was that we would simply use the table that already exists and is used in propertymenu.cpp.  Looks like there is a function elementHasPage() that tells you if an element has a style page, and gotoElement() to jump to that page.  So I just imagined a couple more lines of code somewhere in propertymenu.cpp to make sure "page" gets handled as we want.

Reply:
there is no element `Header` or `Footer` though

Reply:
Tested now and my approach seems to work just fine, and is pretty much self-documenting too

Reply:
I didn't actually try any of this out.  But my assumption was if you right-click on nothing, you get "page" as the element.  Right now the page element maps to the "page" page (!), but it seems that isn't actually being used, because we are building the style menu specially for pages, and the "style" entry opens with no pre-selected page (meaning it gets the last-used page).  So while my original idea might not work, still, we could probably change the page entry in that table in editstyle.cpp to point to header/footer, and then use gotoElement here.

Reply:
Or add header and footer elements and (also) have that dislog being offeren when roght-clickong a heoder/footer?

Reply:
I need to make the inheritance of Ui::EditStyleBase "public" in EditStyle, in order to access EditStly::stackPage and EditStyle::PageHeaderFooter from MuseScore. Or is there any other way to access those members which I'm missing right now.

Reply:
Hmm, I see. And don't have an answer to that for now :-(

Reply:
> Tested now and my approach seems to work just fine, and is pretty much self-documenting too

Were you able to make it work?

Reply:
In ediststyle.cpp, line 396. An invalid test as it turns out now.

Reply:
How about making `MuseScore` friend class of `EditStyle`? It'll preserve `EditStyle`'s private inheritance of `Ui::EditStyleBase`.

Reply:
Hmm, adding `friend class MuseScore;` in editstyle.h's `class EditStyle` indeed seems to do the trick.